File: epi_judge_cpp/gcd.cc
#include "test_framework/generic_test.h"

long long Gcd(long long x, long long y) {
    while (y) {
        long long temp = x;
        x = y;
        y = temp % y;
    }

    return x;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"x", "y"};
    return GenericTestMain(args, "gcd.cc", "gcd.tsv", &Gcd, DefaultComparator{},
                           param_names);
}

File: epi_judge_java/epi/Gcd.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class Gcd {

    @EpiTest(testDataFile = "gcd.tsv")
    public static long GCD(long x, long y) {
        while (y != 0) {
            long temp = y;
            y = x % y;
            x = temp;
        }

        return x;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "Gcd.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}
